llama-index-core/tests/workflow/test_workflow.py

```diff
@@ -37,54 +37,57 @@ async def test_workflow_run(workflow):
     assert result == "Workflow completed"
 
 
-@pytest.mark.asyncio()
-async def test_deprecated_workflow_run_step(workflow):
-    workflow._verbose = True
-
-    # First step
-    result = await workflow.run_step()
-    assert result is None
-    assert not workflow.is_done()
-
-    # Second step
-    result = await workflow.run_step()
-    assert result is None
-    assert not workflow.is_done()
-
-    # Final step
-    result = await workflow.run_step()
-    assert not workflow.is_done()
-    assert result is None
-
-    # Cleanup step
-    result = await workflow.run_step()
-    assert result == "Workflow completed"
-    assert workflow.is_done()
```

Comment:
I think we should probably delete this test and delete `Workflow.run_step()` now? To support this with the re-factor then we could either:

1. Maintain an older version of `_start()` that doesn't contain the new `stepwise` logic and point this `Workflow.run_step()` to it

2. Update the `Worfklow.run_step()` to have a similar refactor to `WorkflowHandler.run_step()`.

Reply:
Maybe we should just delete `run_step()` on the workflow object? (Or at least, delete the implementation and point users towards the updated syntax?)

Reply:
I think we should just delete it at this point tbh. I think if we point it to the updated syntax means that we're changing how this Workflow.run_step() behaves and not sure if we should do that since its already deprecated.

If you and @masci are fine with updating the logic of this and still marking it as deprecated then I'd be happy to adjust accordingly.

Reply:
Alright, I deleted `Workflow.run_step`. I don't think it was worth it to update to point to new code as it would change the functionality of this already deprecated method drastically. As such, keeping it would not really provide no benefits to users who were still using this deprecated method.